modules/org.restlet.test/src/main/java/org/restlet/test/engine/ApplicationContextTestCase.java

```diff
@@ -77,4 +96,10 @@ public void testApplicationContext() throws Exception {
         Representation rep = res.get(MediaType.TEXT_PLAIN);
         assertEquals("WebApiApplication", rep.getText());
     }
+
```

Comment:
thanks for adding a TU!

Reply:
I wonder what is the goal of the test. Should we check whether the application is set or not?

Reply:
@jlouvel 
I have a question.
Here is a description of the context. The connectorService is meant to be [specific to an application](https://github.com/restlet/restlet-framework-java/blob/2.4/modules/org.restlet/src/main/java/org/restlet/service/ConnectorService.java#L35). But the current leads to never call such specific instance of ConnectorService.

Here is the sequence of steps:
- Due to [this fix](https://github.com/restlet/restlet-framework-java/commit/6afed5c70ae36316af4cf471b077d219bc022a00), the current application is set to null after handling the request (in case the sequence is: Component -> Application)
- then the `ServerCall.sendResponse` is called
- the ["current" connectorService](https://github.com/restlet/restlet-framework-java/blob/2.4/modules/org.restlet/src/main/java/org/restlet/engine/adapter/ServerCall.java#L439) is computed and invoked in order to handle the responseEntity before and after bytes are written to the socket.

Because there is no Application as a thread variable, the instance of "current" connectorService is not the one of the Application, it's just a new instance of ConnectorService.

One solution, as proposed by Thomas is to prevent resetting the current application in case it's the "first" one: this is the case for Applications attached to a Component.
Or should we leverage the current context and the [attributes](https://github.com/restlet/restlet-framework-java/blob/2.4/modules/org.restlet/src/main/java/org/restlet/Context.java#L213) map?

Reply:
@thboileau Thanks for the explanation.

As written in the javadocs, usage of Application.getCurrent() should rarely be used, plus ApplicationHelper should be created for one and only one given Application instance helped.

I'm thinking the code should rather be:

        Application.setCurrent(getHelped(getHelped() != null ? getHelped() : Application.getCurrent());

Reply:
You mean:
```java
    Application.setCurrent(getHelped() != null ? getHelped() : Application.getCurrent());
```
?

Reply:
Yes exactly, thanks for spotting the error.